### DISCUSSION TITLE
update() getNewVal() value display delayed one step

### DISCUSSION THREAD

**Discussion by ds5apm** [HomeSpan/HomeSpan#538] [Q&A]
I am using the homespan WindowCovering(door )service, but when I set the target location in the home app, 
the target location value is not immediately reflected, and the previous value is obtained
in getNewVal() at update(). For example, if I set the location to 40 in the home app and then
move the slide to 50, the previous value of 40 is obtained, and if I move it back to 70, the
previous value of 50 is obtained.
Please let me know what I might be doing wrong that is causing this phenomenon.

--code---
  _boolean update(){                             
    if(target->getNewVal()>current->getVal()){      
      LOG1("Raising Shade\n");                     
    } else 
    if(target->getNewVal()<current->getVal()){     
      LOG1("Lowering Shade\n");                    
    } 
    Serial.printf("update()=> Tgt:%d,Crt:%d\n",target->getVal(),current->getVal());      
    return(true);                              
  }_ 
<img width="1309" alt="스크린샷 2023-03-21 오전 5 05 48" src="https://user-images.githubusercontent.com/54817510/226455658-fa14e005-ac45-461a-884c-d1de0d589f7b.png">

**Comment by HomeSpan** [comment]
In your print statement you are printing target->getVal(), not target->getNewVal().

**Reply by ds5apm** [reply]
thank you for your reply
